lightly_studio/tests/core/test_sample.py

```diff
@@ -58,10 +57,6 @@ def test_basic_fields_set(self, mocker: MockerFixture, test_db: Session) -> None
         assert new_image_table.file_name == "sample1.png"
         assert new_image_table.width == 1000
 
-        # Can't set a foreign key to a non-existent id.
-        with pytest.raises(IntegrityError):
-            sample.dataset_id = UUID(int=123456)
```

Comment:
Made dataset_id a non-settable property. Attempts to set the field are caught by mypy.